docs/concepts/multi-chain-proposals.md

```diff
@@ -0,0 +1,693 @@
+    address constant l1CrossDomainMessengerAddress = 0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1;
+
+    // target addresses on Optimism (these addresses would be different for various target chains)
+    address constant v3FactoryTargetAddress = 0x1F98431c8aD98523631AE4a59f267346ea31F984;
+    address constant crossChainAccountTargetAddress = 0xa1dD330d602c32622AA270Ea73d078B803Cb3518;
+
+    // target address for UniSwap DAO governor bravo delegate contract on Ethereum L1
+    address uniswapGovernorBravoDelegatorAddress = 0x408ED6354d4973f66138C91495F2f2FCbd8724C3;
+
+    // function to create a proposal on Ethereum L1 to enable a fee amount on the Uniswap V3 Factory on Optimism.
+    // Upon execution of the proposal, the L1CrossDomainMessenger function sendMessage is called containing calldata for the
+    // CrossChainAccount.forward function on Optimism that will be called containing calldata for the uniswap v3 factory
+    // to enable a fee amount on Uniswap V3 Factory on Optimism
+    function proposeForExecutionOnOptimism() public {
+        // setup calldata for the Optimism uniswap v3Factory.enableFeeAmount call
+        bytes memory _v3FactoryEnableFeeAmounCalldata =
+            abi.encodeWithSelector(IUniswapV3Factory.enableFeeAmount.selector, 10000, 205);
+
+        // encode the calldata for the CrossChainAccount.forward call
+        bytes memory _messageForwardCalldata = abi.encodeWithSelector(
+            ICrossChainAccount.forward.selector, v3FactoryTargetAddress, _v3FactoryEnableFeeAmounCalldata
+        );
+
+        // make the proposal on the L1 side
+        IUniswapGovernorBravoDelegator governor = IUniswapGovernorBravoDelegator(uniswapGovernorBravoDelegatorAddress);
+        address[] memory _targets = new address[](1);
+        uint256[] memory _values = new uint256[](1);
+        string[] memory _signatures = new string[](1);
+        bytes[] memory _calldatas = new bytes[](1);
+        _targets[0] = l1CrossDomainMessengerAddress;
+        _values[0] = 0;
+        _signatures[0] = "sendMessage(address,bytes,uint256)";
+        _calldatas[0] = abi.encode(address(crossChainAccountTargetAddress), _messageForwardCalldata, 500_000);
+        uint256 _proposalId = governor.propose(
+            _targets,
+            _values,
+            _signatures,
+            _calldatas,
+            "Proposal to enable 10000 fee amount of 205 on Uniswap V3 Factory on Optimism"
+        );
+        console2.log("Proposal ID: %d", _proposalId);
+    }
+}
+```
+
+Because 10 networks all use the same CrossChainAccount approach for communication, the only modifications to the above code snippet for those networks would be to change the first 3 address constant definitions. The definitions for all of the OP Stack based networks are below:
+
+| Network       | l1CrossDomainMessengerAddress              | v3FactoryTargetAddress                     | crossChainAccountTargetAddress             |
```

Comment:
nit here: this table is wide on my browser. what about each row having only one cell with all 3 contract addresses?

e.g. 

| Network | Contract addresses |
| --- | --- |
| Base |  `l1CrossDomainMessenger`: &nbsp;`0x866E82a600A1414e583f7F13623F1aC5d58b0Afa`<br/>`v3FactoryTarget`: &nbsp;`0x33128a8fC17869897dcE68Ed026d694621f6FDfD`<br/>`crossChainAccountTarget`: &nbsp;`0x31FAfd4889FA1269F7a13A66eE0fB458f27D72A9` |
| Blast | ... |

not sure I like that all that much more...

what about getting rid of the `Address` suffix on our table headings, and code-formatting each address? maybe that'll get us some of the way there.

Reply:
Done (went with an html table): [769757310e8919e37e364cdbf7f703ef38ffa7c8](https://github.com/ScopeLift/uniswap-docs-fork/pull/1/commits/028fb68f179fec96f2a3296fc54893e43ce0ce60)